ludwig/backend/utils/storage.py

```diff
@@ -0,0 +1,69 @@
+import contextlib
+from typing import Any, Dict, Optional, Union
+
+from ludwig.utils import data_utils
+
+CredInputs = Optional[Union[str, Dict[str, Any]]]
+
+
+DEFAULTS = "defaults"
+ARTIFACTS = "artifacts"
+DATASETS = "datasets"
+CACHE = "cache"
```

Comment:
nit: maybe pull these in from `constants.py` instead of defining them inline? I think some of these might already be there.

Reply:
These I actually want to keep separate, as they're only used internally within this file.

Reply:
Got it!